assets/js/components/user-input/UserInputSelectOptions.js

```diff
@@ -41,19 +41,46 @@ export default function UserInputSelectOptions( { slug, options, max } ) {
 	const values = useSelect( ( select ) => select( CORE_USER ).getUserInputSetting( slug ) || [] );
 	const [ other, setOther ] = useState( values.filter( ( value ) => ! options[ value ] )[ 0 ] || '' );
 	const { setUserInputSetting } = useDispatch( CORE_USER );
+	const inputRef = useRef();
+	const [ disabled, setDisabled ] = useState( false );
 
 	// Need to make sure that dependencies list always has the same number of elements.
 	const dependencies = values.concat( Array( max ) ).slice( 0, max );
 
 	const onClick = useCallback( ( event ) => {
 		const { target } = event;
-		const { value, checked } = target;
+		const { value, checked, name, type, id } = target;
 
 		const newValues = new Set( [ value, ...values ] );
 		if ( ! checked ) {
 			newValues.delete( value );
 		}
 
+		if ( name === `${ slug }-other` && checked === true ) {
+			if ( inputRef.current ) {
+				inputRef.current.inputElement.focus();
+			}
```

Comment:
Previously this used `inputRef.current.handleFocus();`, which doesn't focus the element in my testing and takes a function as an argument according to the types definitions… it seems like it might actually be a prop to handle focus changes, not a function to call on the element.

At any rate, the actual input DOM element is exposed on the attribute `inputElement`, so I changed it here and now focus works as-expected.